core/arch/arm/plat-hikey/spi_test.c

```diff
@@ -28,73 +28,279 @@
 #include <drivers/pl022_spi.h>
 #include <drivers/pl061_gpio.h>
 #include <hikey_peripherals.h>
+#include <io.h>
+#include <kernel/tee_time.h>
 #include <stdint.h>
 #include <trace.h>
+#include <util.h>
 
-/*
- * spi_init() must be run before calling this function.
- *
- * This runs a loopback test by default, so the SPI module will just
- * receive what is transmitted, i.e. 0x01, 0x80, 0x00.
- *
- * In non-loopback mode, the transmitted value will elicit a readback of
- * the measured value from the ADC chip on the Linksprite 96Boards
- * Mezzanine card [1], which can be connected to either a sliding
- * rheostat [2] or photoresistor [3].
- *
- * [1] http://linksprite.com/wiki/index.php5?title=Linker_Mezzanine_card_for_96board
- * [2] http://learn.linksprite.com/96-board/sliding-rheostat
- * [3] http://learn.linksprite.com/96-board/photoresistor
- */
-void spi_test(void)
+#define PL022_STAT	0x00C
+#define PL022_STAT_BSY	SHIFT_U32(1, 4)
+
+static void spi_cs_callback(enum gpio_level value)
 {
+	static bool is_init = false;
```

Comment:
@vchong, if you don't feel the boolean null value is not explicit enough as a 'false' identifier, maybe you can switch to a `static int inited;`: i think everyone expect such default initializes to 0.